core/controllers/editor.py

```diff
@@ -960,30 +962,43 @@ def post(self, exploration_id):
                 'Invalid filename extension: it should have '
                 'one of the following extensions: %s' % allowed_formats)
 
-        temp_suffix = '.' + extension
-        with tempfile.NamedTemporaryFile(suffix=temp_suffix) as f:
-            f.write(raw)
-            try:
-                audio = mutagen.File(f.name)
-            except mutagen.MutagenError:
-                raise self.InvalidInputException('Audio not recognized '
-                                                 'as a %s file' % extension)
-            f.close()
-
+        tempbuffer = StringIO.StringIO()
+        tempbuffer.write(raw)
```

Comment:
Wow, that is ... odd. That really goes against my understanding of how things work, and I must say, I'm pretty confused.

Let's keep it as you have it, but add a lengthy comment above `del audio` explaining what's going on. I'll try and dig a bit into it and see if I can figure it out.